src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,25 @@
 package mate.academy;
 
+import java.util.concurrent.CopyOnWriteArrayList;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final CopyOnWriteArrayList<EventListener> queue = new CopyOnWriteArrayList<>();
 
+    public void registerListener(EventListener listener) {
+        queue.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        queue.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : queue) {
+            listener.onEvent(event);
+        }
     }
 
     public void shutdown() {
```

Comment:
The shutdown method currently only prints a message to the console. If there are any resources or threads that need to be cleaned up or stopped when shutting down the EventManager, this would be the place to do it. Make sure that all resources are properly released to avoid resource leaks.